openpectus/lang/exec/tags_impl.py

```diff
@@ -14,16 +14,28 @@ class ReadingTag(Tag):
     def __init__(self, name: str, unit: str | None = None) -> None:
         super().__init__(name, value=0.0, unit=unit, direction=TagDirection.INPUT)
 
+    def set_value(self, val, *args, **kwargs):
```

Comment:
I think we should try to catch this earlier, probably in validation. set_value may be called in every tick so raising an error here may result in too many log messages to be helpful.

Reply:
Good point regarding the possible error-flood. When you say validation do you then mean initial UOD validation? I propose you take over on this issue and find the proper place to make the check :-).